ferrocene/doc/internal-procedures/src/working-with-the-ci.rst

```diff
@@ -0,0 +1,226 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You do not need to download or use `Docker Desktop
+<https://www.docker.com/products/docker-desktop/>`_ with Ferrocene. If you
+choose to use Docker Desktop, install it as directed on the Website and skip
+the rest of this section.
+
+On Linux, install Docker and `docker-buildx` from your package manager, for
+example, `apt install docker.io docker-buildx`. On some distributions, you may
+need to start the service with `systemctl enable --now docker`.
+
+On MacOS, you can use the Docker CLI connected to a Lima host:
+
+.. code-block:: bash
+
+   brew install Lima docker
+   limactl start template://docker
+   docker context create lima-docker --docker "host=unix://{{.Dir}}/sock/docker.sock"
+   docker context use lima-docker
+   docker run --rm hello-world
+
+On Windows, it's generally recommended to do CI imitation within WSL2. Enable
+WSL2 from an administrator Powershell if you haven't done so already, then
+`ubuntu-24.04`,. from there, follow the Linux instructions above:
+
+.. code-block:: Powershell
+
+   wsl --install
+   wsl --install -d Ubuntu-24.04
+
+
+Installing Python
+^^^^^^^^^^^^^^^^^
+
+Ferrocene's CI pins Python 3.12.x currently. In general, any modern Python
+should be sufficient for local CI imitation.
+
+On Linux, install Python with your distribution's package manager, for example:
+
+.. code-block:: bash
+
+   apt install python
+
+On macOS, Python is already installed if you have a working XCode toolchain or
+XCode Developer Tools installed, for example:
+
+.. code-block:: bash
+   
+   xcode-select --install
+
+
+.. warning::
+   
+   Do not install Python or ``pip`` via Homebrew, if you have already done so,
+   remove them.
+
+On Windows, run ``winget install Python.Python.3.12`` or get it from the
+`website <https://www.python.org/downloads/windows/>`_.
+
+
+Using the Python virtual environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ferrocene uses ``uv`` to manage our Python dependencies. There are a number of
+ways to install ``uv <https://github.com/astral-sh/uv>``_, but we recommend the
+``curl`` (Linux, MacOS) or ``powershell``  (Windows) methods:
+
+.. code-block:: bash
+
+   curl -LsSf https://astral.sh/uv/install.sh | sh
+
+.. code-block:: powershell
+
+   powershell -c "irm https://astral.sh/uv/install.ps1 | iex"
+
+Follow the onscreen instructions post-install to ensure `uv` is present on your
+``$PATH``.
+
+Set up the virtual environment:
+
+.. code-block:: bash
+
+   uv venv
+   source .venv/bin/activate
+   uv pip sync requirements.txt
+
+You need to source the virtual enviroment each time you wish to use it:
+
+.. code-block:: bash
+
+   source .venv/bin/activate
+
+
+Using the CI ``config.toml``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To create the ``config.toml`` used by the CI, set ``FERROCENE_HOST`` to your host triple,
+then run the ``configure.sh``:
+
+
+.. code-block:: bash
+
+   CI=true FERROCENE_HOST=aarch64-unknown-linux-gnu ./ferrocene/ci/configure.sh 
+
```

Comment:
We should add a warning noting that this will override your existing configuration file, and that configuration files meant for CI are not suitable for local use. Another option is to remove the `CI=true` and have the users stumble into the error it prints.